### PR TITLE
tests: adding test for removable-media interface

### PR DESCRIPTION
**Pull request by sergiocazzolato**

### CODE REVIEW

File: tests/main/interfaces-removable-media/task.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+execute: |
+    echo "The interface is not connected by default"
+    snap interfaces | MATCH "$DISCONNECTED_PATTERN"
+
+    echo "When the interface is connected"
+    snap connect test-snapd-removable-media:removable-media
+    snap interfaces | MATCH "$CONNECTED_PATTERN"
+
+    echo "Then the snap is able to read inside /run"
+    test-snapd-removable-media.sh -c "test -d /run/snapd"
```
Comment (zyga) [inline tests/main/interfaces-removable-media/task.yaml]: The interface doesn't talk about `/run/snapd` in any way. Did you find this documented as such somewhere?

The only thing that the interface grants is `/media` or alternatively `/run/media` for read access (to enumerate) and `/media/*/` or alternatively `/run/media/*/` for read-write access (to get the actual data).

File: tests/main/interfaces-removable-media/task.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+summary: Ensure that the removable-media interface works.
+
+details: |
+    The removable-media interface allows to access to removable storage filesystems.
+environment:
+    TESTDIR: '/media/testdir'
```
Comment (zyga) [inline tests/main/interfaces-removable-media/task.yaml]: Can you please test the `/run/media/foo` variant as well? You can drop the `TESTFILE` variable and just use `TESTDIR/file` for simplicity.

Comment (bboozzoo) [reply tests/main/interfaces-removable-media/task.yaml]: `/media` and `/run/media` split is a bit confusing as I stumbled upon this in #4285. There's a `--enable-merged-usr` snap-confine configure option, that results in either `/media` or `/run/media` being mounted into the child's namespace. IIRC there was some incosistency between Fedora/Arch/OpenSUSE on this matter. I think Arch ended up with `/media`, what made me doubt that the actual location is related to usr-split.

File: tests/main/interfaces-removable-media/task.yaml (hunk 1 above)
Comment (bboozzoo) [reply tests/main/interfaces-removable-media/task.yaml]: I think it also grants read on `/run`.

File: tests/lib/snaps/test-snapd-removable-media/meta/snap.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+name: test-snapd-removable-media
+version: 1.0
+summary: Basic removable-media snap
+description: A basic snap which access to removable storage filesystems
+
+apps:
+  sh:
+    command: ../../../bin/sh
```
Comment (niemeyer) [inline tests/lib/snaps/test-snapd-removable-media/meta/snap.yaml]: Ouch! This ought to be invalid. It should be an absolute path instead.

@chipaca Did we get that fixed?  I recall we talked about it recently, but don't know if we acted on it.

File: tests/main/interfaces-removable-media/task.yaml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+prepare: |
+    . $TESTSLIB/snaps.sh
+    install_local test-snapd-removable-media
+
+    if ! [ -d /media ]; then
+        mkdir -p /media
+        touch /media/fake
+    fi
+    mkdir -p /media/testdir
```
Comment (zyga) [inline tests/main/interfaces-removable-media/task.yaml]: Just a remark that this test is a little unrealistic in that in real life /media would show _either_ /media or /run/media, depending on how snap-confine is configured at build time. Still, this is sufficient for the moment and we can improve it once that distinction is handled automatically by snap-confine.

**Pull request merged**